src/lib/sw.js

```diff
@@ -104,56 +104,19 @@ workboxRouting.registerRoute(
 );
 
 /**
- * This is a special handler for requests without a trailing "/". These requests _should_ go to
- * the network (so that we can match web.dev's redirects.yaml file) but fallback to the normalized
- * version of the request (e.g., "/foo" => "/foo/").
+ * Match fetches for partials, for SPA requests. Matches requests like:
+ *   - /index.json
+ *   - /foo-bar/index.json
+ *   - /foo-bar/many/parts/test.json
  */
-const untrailedContentPathRe = new RegExp('^(/[\\w-]+)+$');
-
-workboxRouting.registerRoute(
-  matchSameOriginRegExp(untrailedContentPathRe),
-  async ({url, event}) => {
-    const {pathname} = url;
-
-    // First, check if there's actually a partial node in the cache already. If so, there's no need
-    // to do a real network fetch, as we know the modified request will be satisfied.
-    const cachedResponse = await caches.match(
-      new Request(pathname + '/index.json'),
-    );
-    if (!cachedResponse) {
-      // If there's not, then try the network.
-      try {
-        return await fetch(event.request);
-      } catch (e) {
-        // If fetch fails, just redirect below.
-      }
-    }
-
-    // Either way, redirect to the updated Location.
-    const headers = new Headers();
-    headers.append('Location', `${pathname}/${url.search}`);
-    const redirectResponse = new Response('', {
-      status: 301,
-      headers,
-    });
-    return redirectResponse;
-  },
-);
-
-/**
- * Match fetches for patials, for SPA requests. Matches "/foo-bar/index.json" and
- * "/foo-bar/many/parts/index.json", for partial SPA requests.
- */
-const partialPathRe = new RegExp('^/([\\w-]+/)*index\\.json$');
+const partialPathRe = new RegExp('^/([\\w-]+/)*\\w+\\.json$');
```

Comment:
what happens if the site requests a `.json` file that isn't related to partials. Will this send it to the network?